### PR TITLE
[SMALLFIX] Improved Mesos framework logging

### PR DESCRIPTION
**Pull request by efeller**
Make use of logger in the Mesos framework

### CODE REVIEW

File: integration/mesos/src/main/java/alluxio/mesos/AlluxioFramework.java [hunk 1]
```diff
@@ -65,43 +69,43 @@ public AlluxioScheduler() {}
...
     public void error(SchedulerDriver driver, String message) {
-      System.out.println("Error: " + message);
+      LOG.error(String.format("Error: %s", message));
     }
 
     @Override
     public void executorLost(SchedulerDriver driver, Protos.ExecutorID executorId,
         Protos.SlaveID slaveId, int status) {
-      System.out.println("Executor " + executorId.getValue() + " was lost.");
+      LOG.info(String.format("Executor %s was lost", executorId.getValue()));
```
Comment (jsimsa) [inline integration/mesos/src/main/java/alluxio/mesos/AlluxioFramework.java]: `LOG.info("Executor {} was lost", executorId.getValue())`

File: integration/mesos/src/main/java/alluxio/mesos/AlluxioFramework.java [hunk 2]
```diff
@@ -65,43 +69,43 @@ public AlluxioScheduler() {}
...
     }
 
     @Override
     public void frameworkMessage(SchedulerDriver driver, Protos.ExecutorID executorId,
-        Protos.SlaveID slaveId, byte[] data) {
-      System.out.println("Executor: " + executorId.getValue() + ", " + "Slave: "
-          + slaveId.getValue() + ", " + "Data: " + Arrays.toString(data) + ".");
+                                 Protos.SlaveID slaveId, byte[] data) {
+      LOG.info(String.format("Executor: %s, slave: %s, data: %s",
```
Comment (jsimsa) [inline integration/mesos/src/main/java/alluxio/mesos/AlluxioFramework.java]: ditto

File: integration/mesos/src/main/java/alluxio/mesos/AlluxioFramework.java [hunk 3]
```diff
@@ -65,43 +69,43 @@ public AlluxioScheduler() {}
...
+          executorId.getValue(), slaveId.getValue(), Arrays.toString(data)));
     }
 
     @Override
     public void offerRescinded(SchedulerDriver driver, Protos.OfferID offerId) {
-      System.out.println("Offered " + offerId.getValue() + " rescinded.");
+      LOG.info(String.format("Offered %s rescinded", offerId.getValue()));
```
Comment (jsimsa) [inline integration/mesos/src/main/java/alluxio/mesos/AlluxioFramework.java]: ditto

File: integration/mesos/src/main/java/alluxio/mesos/AlluxioFramework.java [hunk 4]
```diff
@@ -65,43 +69,43 @@ public AlluxioScheduler() {}
...
     }
 
     @Override
     public void registered(SchedulerDriver driver, Protos.FrameworkID frameworkId,
-        Protos.MasterInfo masterInfo) {
-      System.out.println("Registered framework " + frameworkId.getValue() + " with master "
-          + masterInfo.getHostname() + ":" + masterInfo.getPort() + ".");
+                           Protos.MasterInfo masterInfo) {
+      LOG.info(String.format("Registered framework %s with master %s:%d",
```
Comment (jsimsa) [inline integration/mesos/src/main/java/alluxio/mesos/AlluxioFramework.java]: ditto

File: integration/mesos/src/main/java/alluxio/mesos/AlluxioFramework.java [hunk 5]
```diff
@@ -65,43 +69,43 @@ public AlluxioScheduler() {}
...
+          frameworkId.getValue(), masterInfo.getHostname(), masterInfo.getPort()));
     }
 
     @Override
     public void reregistered(SchedulerDriver driver, Protos.MasterInfo masterInfo) {
-      System.out.println("Registered framework with master " + masterInfo.getHostname() + ":"
-          + masterInfo.getPort() + ".");
+      LOG.info(String.format("Registered framework with master %s:%d",
```
Comment (jsimsa) [inline integration/mesos/src/main/java/alluxio/mesos/AlluxioFramework.java]: ditto

File: integration/mesos/src/main/java/alluxio/mesos/AlluxioFramework.java [hunk 6]
```diff
@@ -113,6 +117,10 @@ public void resourceOffers(SchedulerDriver driver, List<Protos.Offer> offers) {
       long workerMem =
           Configuration.getBytes(Constants.INTEGRATION_WORKER_RESOURCE_MEM) / Constants.MB;
 
+      LOG.info(String.format("Master launched %b, master count %d, "
```
Comment (jsimsa) [inline integration/mesos/src/main/java/alluxio/mesos/AlluxioFramework.java]: ditto

File: integration/mesos/src/main/java/alluxio/mesos/AlluxioFramework.java (hunk 1 above)
Comment (jsimsa) [inline integration/mesos/src/main/java/alluxio/mesos/AlluxioFramework.java]: see http://www.slf4j.org/faq.html#logging_performance for details

File: integration/mesos/src/main/java/alluxio/mesos/AlluxioFramework.java (hunk 6 above)
Comment (jsimsa) [inline integration/mesos/src/main/java/alluxio/mesos/AlluxioFramework.java]: ditto

File: integration/mesos/src/main/java/alluxio/mesos/AlluxioFramework.java (hunk 1 above)
Comment (jsimsa) [inline integration/mesos/src/main/java/alluxio/mesos/AlluxioFramework.java]: also, apply this to all of you LOG statements

File: integration/mesos/src/main/java/alluxio/mesos/AlluxioFramework.java [hunk 7]
```diff
@@ -65,43 +69,43 @@ public AlluxioScheduler() {}
...
-      System.out.println("Offered " + offerId.getValue() + " rescinded.");
+      LOG.info(String.format("Offered %s rescinded", offerId.getValue()));
     }
 
     @Override
     public void registered(SchedulerDriver driver, Protos.FrameworkID frameworkId,
-        Protos.MasterInfo masterInfo) {
-      System.out.println("Registered framework " + frameworkId.getValue() + " with master "
-          + masterInfo.getHostname() + ":" + masterInfo.getPort() + ".");
+                           Protos.MasterInfo masterInfo) {
```
Comment (jsimsa) [inline integration/mesos/src/main/java/alluxio/mesos/AlluxioFramework.java]: fix indentation

File: integration/mesos/src/main/java/alluxio/mesos/AlluxioFramework.java [hunk 8]
```diff
@@ -65,43 +69,43 @@ public AlluxioScheduler() {}
...
-      System.out.println("Executor " + executorId.getValue() + " was lost.");
+      LOG.info(String.format("Executor %s was lost", executorId.getValue()));
     }
 
     @Override
     public void frameworkMessage(SchedulerDriver driver, Protos.ExecutorID executorId,
-        Protos.SlaveID slaveId, byte[] data) {
-      System.out.println("Executor: " + executorId.getValue() + ", " + "Slave: "
-          + slaveId.getValue() + ", " + "Data: " + Arrays.toString(data) + ".");
+                                 Protos.SlaveID slaveId, byte[] data) {
```
Comment (jsimsa) [inline integration/mesos/src/main/java/alluxio/mesos/AlluxioFramework.java]: fix indentation

File: integration/mesos/src/main/java/alluxio/mesos/AlluxioFramework.java [hunk 9]
```diff
@@ -43,13 +45,15 @@
...
 public class AlluxioFramework {
+  private static final Logger LOG = LoggerFactory.getLogger(Constants.LOGGER_TYPE);
 
   /**
    * Creates a new {@link AlluxioFramework}.
    */
   public AlluxioFramework() {}
 
   static class AlluxioScheduler implements Scheduler {
+    private static final Logger LOG = LoggerFactory.getLogger(Constants.LOGGER_TYPE);
```
Comment (jsimsa) [inline integration/mesos/src/main/java/alluxio/mesos/AlluxioFramework.java]: nit: you could just reuse the outer class logger

File: integration/mesos/src/main/java/alluxio/mesos/AlluxioMasterExecutor.java [hunk 10]
```diff
@@ -89,19 +93,19 @@ public void run() {
 
   @Override
   public void registered(ExecutorDriver driver, Protos.ExecutorInfo executorInfo,
-      Protos.FrameworkInfo frameworkInfo, Protos.SlaveInfo slaveInfo) {
-    System.out.println("Registered executor " + executorInfo.getName() + " with "
-        + slaveInfo.getHostname() + " through framework " + frameworkInfo.getName() + ".");
+                         Protos.FrameworkInfo frameworkInfo, Protos.SlaveInfo slaveInfo) {
```
Comment (jsimsa) [inline integration/mesos/src/main/java/alluxio/mesos/AlluxioMasterExecutor.java]: fix indentation

File: integration/mesos/src/main/java/alluxio/mesos/AlluxioWorkerExecutor.java [hunk 11]
```diff
@@ -89,19 +92,19 @@ public void run() {
 
   @Override
   public void registered(ExecutorDriver driver, Protos.ExecutorInfo executorInfo,
-      Protos.FrameworkInfo frameworkInfo, Protos.SlaveInfo slaveInfo) {
-    System.out.println("Registered executor " + executorInfo.getName() + " with "
-        + slaveInfo.getHostname() + " through framework " + frameworkInfo.getName() + ".");
+                         Protos.FrameworkInfo frameworkInfo, Protos.SlaveInfo slaveInfo) {
```
Comment (jsimsa) [inline integration/mesos/src/main/java/alluxio/mesos/AlluxioWorkerExecutor.java]: fix indentation

File: integration/mesos/src/main/java/alluxio/mesos/OfferUtils.java [hunk 12]
```diff
@@ -15,13 +15,16 @@
...
+import org.slf4j.LoggerFactory;
 
 /**
  * Mesos framework offer utils.
  */
 public final class OfferUtils {
+  private static final Logger LOG = LoggerFactory.getLogger(Constants.LOGGER_TYPE);
   private OfferUtils() {} // prevent instantiation
-
+
```
Comment (jsimsa) [inline integration/mesos/src/main/java/alluxio/mesos/OfferUtils.java]: remove trailing whitespace

**Comment by jsimsa** [comment]
@efeller thank you for your contribution, left some minor comments for you

**Comment by efeller** [comment]
@jsimsa Thanks for the review. I will try to address the comments today.

**Comment by efeller** [comment]
@jsimsa done, please let me know if I missed something. :-)

**Comment by jsimsa** [comment]
LGTM, thanks for the cleanup

**Pull request merged**